quixstreams/state/rocksdb/partition.py

```diff
@@ -103,28 +99,71 @@ def _changelog_recover_flush(self, changelog_offset: int, batch: WriteBatch):
         )
         self.write(batch)
 
+    def _should_skip_changelog(
+        self, headers: Dict[str, bytes], committed_offset: int
+    ) -> bool:
+        """
+        Determine whether the changelog update should be skipped.
+
+        :param headers: changelog message headers
+        :param committed_offset: latest committed offset of the source topic partition
+        :return: True if update should be skipped, else False.
+        """
+        # Parse the processed topic-partition-offset info from the changelog message
+        # headers to determine whether the update should be applied or skipped.
+        # It can be empty if the message was produced by the older version of the lib.
+        processed_offset_header = headers.get(
+            CHANGELOG_PROCESSED_OFFSET_MESSAGE_HEADER, b"null"
+        )
+        processed_offset = json_loads(processed_offset_header)
```

Comment:
Does this need to be JSON anymore due to your previous adjustments to this? Might be a little faster to not use JSON

Reply:
The JSON is used here to simplify the debugging in case you need to examine the produced messages using some Kafka UI.
I also checked the performance, and the `orjson` lib is as fast as serializing ints to bytes using `struct`.